### PR TITLE
Deserialize Hearthstone JSON file

### PR DESCRIPTION
**Pull request by vecum0814**
This revision includes:
- Deserialize Hearthstone JSON file
  - Using crate 'serde'
- Update a download link of 'grcov'
- Update Codecov's GitHub Action 'codecov-action' to v3

### CODE REVIEW

File: src/loaders/json_reader.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+use std::path::Path;
+use std::fs::File;
+
+use serde::{Deserialize};
+
+
```
Comment (utilForever) [inline src/loaders/json_reader.rs]:
```suggestion
use serde::Deserialize;

```

File: src/loaders/json_reader.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+struct Address {
+    street: String,
+    city: String,
+    country: String,
+}
+
```
Comment (utilForever) [inline src/loaders/json_reader.rs]: These structures seem unnecessary.

File: src/loaders/json_reader.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    referenced_tags: Option<Vec<String>>,
+    set: String,
+    tech_level: Option<u8>,
+    spell_school: Option<String>,
+    text: Option<String>,
+    r#type: String,
+}
+
+
+
```
Comment (utilForever) [inline src/loaders/json_reader.rs]:
```suggestion
```

Please remove exceeded blank lines.

File: src/loaders/json_reader.rs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        //println!("dbfId: {}", card.dbf_id)
+    }
+
+    for card in &battleground_cards {
+ 
+        println!("Card Name: {}", card.name)
+    }
+    println!("{}", battleground_cards.len())
+
+}
```
Comment (utilForever) [inline src/loaders/json_reader.rs]:
```suggestion
fn main() {
    let path = Path::new("/Users/raylee/Rust/json_reader/src/cards_kor.json");
    let file = File::open(path).expect("file not found");
    let mut cards = Vec::new();

    let raw_cards: Vec<Card> = serde_json::from_reader(file).expect("Invalid format");
    
    for card in raw_cards.iter() {
        if card.battlegrounds_premium_dbf_id.is_some() {
            cards.push(card);
        }
    }

    for card in battleground_cards.iter() {
        println!("Card name: {}", card.name)
    }
    
    println!("{}", battleground_cards.len())
}
```

File: src/loaders/json_reader.rs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    spell_school: Option<String>,
+    text: Option<String>,
+    r#type: String,
+}
+
+
+
+
+fn main() {
+    let json_file_path = Path::new("/Users/raylee/Rust/json_reader/src/cards_kor.json");
```
Comment (utilForever) [inline src/loaders/json_reader.rs]: You should use relative path instead of absolute path.

**Review by utilForever — CHANGES_REQUESTED** [review]
You should change some codes.
Also, you should add unit test code to check the logic.

**Comment by utilForever** [comment]
I pushed three commits.
- Apply code format by rustfmt
- Update a download link of 'grcov'
- Update Codecov's GitHub Action 'codecov-action' to v3

Now, CIs are worked! 😄

File: src/loaders/json_reader.rs (hunk 1 above)
Comment (vecum0814) [reply src/loaders/json_reader.rs]: Applied the suggested modifications

File: src/loaders/json_reader.rs (hunk 2 above)
Comment (vecum0814) [reply src/loaders/json_reader.rs]: Extracted the unnecessary structures.

File: src/loaders/json_reader.rs (hunk 3 above)
Comment (vecum0814) [reply src/loaders/json_reader.rs]: Removed the unnecessary blank lines.

File: src/loaders/json_reader.rs (hunk 4 above)
Comment (vecum0814) [reply src/loaders/json_reader.rs]: Manipulated the names of suggested variables.

File: src/loaders/json_reader.rs (hunk 5 above)
Comment (vecum0814) [reply src/loaders/json_reader.rs]: Changed the absolute path into a relative path.

**Review by utilForever — APPROVED** [review]
LGTM now. Thanks for your work! 🎉

**Review by overthestream — APPROVED** [review]

**Pull request merged**